### PR TITLE
gl_shader_cache: Rework shader cache and store texture arrays

### PR DESCRIPTION
**Pull request by ReinUsesLisp**
Instead of pre-specializing shaders and then post-specializing them, drop the later and only "specialize" the shader while decoding it. This reduces the complexity of the shader cache significantly.

While we are changing code that invalidates the disk shader cache, do some maintenance to the code related to it.

* This invalidates the disk shader cache.
* This fixes reflections in LM3 when the shader cache is enabled.

### CODE REVIEW

File: src/video_core/renderer_opengl/gl_shader_decompiler.cpp [hunk 1]
```diff
@@ -56,6 +57,25 @@ using TextureIR = std::variant<TextureOffset, TextureDerivates, TextureArgument>
 constexpr u32 MAX_CONSTBUFFER_ELEMENTS =
     static_cast<u32>(Maxwell::MaxConstBufferSize) / (4 * sizeof(float));
 
+std::string_view CommonDeclarations = R"(#define ftoi floatBitsToInt
```
Comment (lioncash) [inline src/video_core/renderer_opengl/gl_shader_decompiler.cpp]: This can be constexpr

File: src/video_core/shader/registry.h [hunk 2]
```diff
@@ -82,22 +105,33 @@ class ConstBufferLocker {
         return bound_buffer;
     }
 
+    /// Returns compute information from this shader
+    const GraphicsInfo& GetGraphicsInfo() const {
```
Comment (lioncash) [inline src/video_core/shader/registry.h]: These should probably be defined in the cpp file, or have `common/assert.h` included.

Comment (ReinUsesLisp) [reply src/video_core/shader/registry.h]: The commentary is also wrong.

**Comment by BSoDGamingYT** [comment]
This PR also fixed graphical corruption when moving room to room in Luigi's Mansion 3 on OpenGL

**Pull request merged**